ocs_ci/ocs/ui/pvc_ui.py

```diff
@@ -87,28 +98,79 @@ def verify_pvc_ui(self, pvc_size, access_mode, vol_mode, sc_type):
         if (
             sc_type
             in (
-                "ocs-storagecluster-ceph-rbd-thick",
-                "ocs-storagecluster-ceph-rbd",
+                constants.DEFAULT_STORAGECLASS_RBD_THICK,
+                constants.DEFAULT_STORAGECLASS_RBD,
             )
             and (access_mode == "ReadWriteOnce")
         ):
             pvc_vol_mode_new = f"{vol_mode}"
             self.check_element_text(expected_text=pvc_vol_mode_new)
             logger.info(f"Verifying volume mode : {pvc_vol_mode_new}")
 
-    def delete_pvc_ui(self, pvc_name):
+    def pvc_resize_ui(self, project_name, pvc_name, new_size):
+        """
+        Resizing pvc via UI
+
+        Args:
+            project_name (str): name of test project
+            pvc_name (str): the name of pvc
+            new_size (int): the new size of pvc (GB)
+
+        """
+
+        self.navigate_persistentvolumeclaims_page()
+
+        logger.info(f"Search and Select test project {project_name}")
+        self.do_click(self.pvc_loc["pvc_project_selector"])
+        self.do_send_keys(self.pvc_loc["search-project"], text=project_name)
+        self.do_click((f"//a[normalize-space()='{project_name}']", By.XPATH))
+
+        logger.info(f"Search for {pvc_name} inside test project {project_name}")
+        self.do_send_keys(self.pvc_loc["search_pvc"], text=pvc_name)
+
+        logger.info(f"Go to PVC {pvc_name} Page")
+        self.do_click(self.pvc_loc[pvc_name])
+
+        logger.info("Checking status of Pvc")
+        self.wait_for_element(
+            (
+                "dd[data-test-id='pvc-status'] span[data-test='status-text']",
+                By.CSS_SELECTOR,
+            )
+        )
+
+        logger.info("Click on Actions")
+        self.do_click(self.pvc_loc["pvc_actions"])
+
+        logger.info("Click on Expand PVC from dropdown options")
+        self.do_click(self.pvc_loc["expand_pvc"])
+
+        logger.info("Clear the size of existing pvc")
+        self.do_clear(self.pvc_loc["resize-value"])
+
+        logger.info("Enter the size of new pvc")
+        self.do_send_keys(self.pvc_loc["resize-value"], text=new_size)
+
+        logger.info("Click on Expand Button")
+        self.do_click(self.pvc_loc["expand-btn"])
+
+    def delete_pvc_ui(self, pvc_name, project_name):
         """
         Delete pvc via UI
 
-        pvc_name (str): Name of the pvc
+        Args:
+            pvc_name (str): Name of the pvc
+            project_name (str): name of test project
 
         """
         self.navigate_persistentvolumeclaims_page()
 
-        logger.info("Select openshift-storage project")
+        logger.info(f"Select test project {project_name}")
         self.do_click(self.pvc_loc["pvc_project_selector"])
-        self.do_click(self.pvc_loc["select_openshift-storage_project"])
+        self.do_send_keys(self.pvc_loc["search-project"], text=project_name)
+        self.do_click((f"//a[normalize-space()='{project_name}']", By.XPATH))
```

Comment:
please move this element to views.py

Reply:
Not sure if we can access project_name in views.py file, because it is dynamic and name of the project changes every time, I will then have to create an object of TestPvcUserInterface class of test_pvc_ui file inside views.py file. Please suggest a suitable way to do so.

Reply:
@OdedViner Done